src/InContextSignup.js

```diff
@@ -104,13 +104,13 @@ export class InContextSignup {
     }
 
     onIncontextSignupDismissed (options = { shouldHideTooltip: true }) {
-        this.permanentlyDismissedAt = new Date().getTime()
-        this.device.deviceApi.notify(new SetIncontextSignupPermanentlyDismissedAtCall({ value: this.permanentlyDismissedAt }))
-        this.device.firePixel({pixelName: 'incontext_dismiss_persisted'})
         if (options.shouldHideTooltip) {
             this.device.removeAutofillUIFromPage()
             this.device.deviceApi.notify(new CloseAutofillParentCall(null))
         }
+        this.permanentlyDismissedAt = new Date().getTime()
+        this.device.deviceApi.notify(new SetIncontextSignupPermanentlyDismissedAtCall({ value: this.permanentlyDismissedAt }))
+        this.device.firePixel({pixelName: 'incontext_dismiss_persisted'})
```

Comment:
@alistairjcbrown Here's the fix. Now we remove the styles before setting in-context data. I've checked browser and extension and it seems to be working splendidly, but do let me know if you have concerns.

Oh, and now that we have fixed the test for this, should this ever regress we'd know right away 💪.